go/pkg/context/detached.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018 Iguazio
```

Comment:
maybe not needed?
see https://pkg.go.dev/golang.org/x/tools@v0.5.0/internal/xcontext

Reply:
This is indeed the same, however I'm having trouble using it on MacOS due to some build constraints in the `tools` package.
I'll add a TODO to use this package but will defer it for now.